File: desktop/src/shared/creatorSearch.ts
import type { ApiResult, MatchPage, Pagination, PlanCreate } from './match';
export type CreatorSearchStatus = 'queued'|'running'|'stopping'|'stopped'|'completed'|'partial'|'failed';
export type CreatorSearchStage = 'planning'|'discovery'|'profiles'|'emails'|'screening'|'deep_match'|'ranking'|'complete';
export interface CreatorSearch {
  id:string;activity_id:string;plan_id:string;query_id:string|null;evaluation_id:string|null;parent_search_id:string|null;
  status:CreatorSearchStatus;stage:CreatorSearchStage;
  counts:{discovered:number;profile_ready:number;profile_reused:number;profile_failed:number;email_available:number;email_missing:number;email_failed:number;evaluated:number;matched:number};
  stop_requested:boolean;retryable:boolean;outcome_unknown:boolean;error_code:string|null;created_at:string;updated_at:string;
}
export interface CreatorSearchPerson {
  candidate_id:string;creator_id:string;platform:'youtube'|'x'|'twitch'|'instagram';
  profile_status:'pending'|'running'|'ready'|'reused'|'failed';email_status:'pending'|'running'|'available'|'missing'|'failed';
  analysis_job_id:string|null;profile_error_code:string|null;email_error_code:string|null;
}
export interface CreatorSearchAccepted {search_id:string;status:CreatorSearchStatus}
export interface CreatorSearchAPI {
  creatorSearches(input:Pagination&{activityId:string}):Promise<ApiResult<MatchPage<CreatorSearch>>>;
  createCreatorSearch(input:{activityId:string;data:PlanCreate;idempotencyKey:string}):Promise<ApiResult<CreatorSearchAccepted>>;
  creatorSearch(id:string):Promise<ApiResult<CreatorSearch>>;
  creatorSearchPeople(input:Pagination&{id:string}):Promise<ApiResult<MatchPage<CreatorSearchPerson>>>;
  stopCreatorSearch(input:{id:string;idempotencyKey:string}):Promise<ApiResult<CreatorSearchAccepted>>;
  retryCreatorSearch(input:{id:string;data:{acknowledge_unknown?:boolean};idempotencyKey:string}):Promise<ApiResult<CreatorSearchAccepted>>;
  appendCreatorSearch(input:{id:string;idempotencyKey:string}):Promise<ApiResult<CreatorSearchAccepted>>;
}

File: backend/app/schemas/creator_search.py
from datetime import datetime
from typing import Literal
from uuid import UUID
from pydantic import BaseModel
from app.schemas.activity import StrictModel
from app.schemas.discovery_planning import PlanCreate


class CreatorSearchCreate(PlanCreate):
    mode: Literal["discover"] = "discover"


class CreatorSearchRetry(StrictModel):
    acknowledge_unknown: bool = False


class CreatorSearchEmpty(StrictModel):
    pass


class CreatorSearchAccepted(BaseModel):
    search_id: UUID
    status: str


class CreatorSearchCounts(BaseModel):
    discovered: int = 0
    profile_ready: int = 0
    profile_reused: int = 0
    profile_failed: int = 0
    email_available: int = 0
    email_missing: int = 0
    email_failed: int = 0
    evaluated: int = 0
    matched: int = 0


class CreatorSearchView(BaseModel):
    id: UUID
    activity_id: UUID
    plan_id: UUID
    query_id: UUID | None
    evaluation_id: UUID | None
    parent_search_id: UUID | None
    status: Literal[
        "queued", "running", "stopping", "stopped", "completed", "partial", "failed"
    ]
    stage: Literal[
        "planning",
        "discovery",
        "profiles",
        "emails",
        "screening",
        "deep_match",
        "ranking",
        "complete",
    ]
    counts: CreatorSearchCounts
    stop_requested: bool
    retryable: bool
    outcome_unknown: bool
    error_code: str | None
    created_at: datetime
    updated_at: datetime


class CreatorSearchUnitView(BaseModel):
    candidate_id: UUID
    creator_id: UUID
    platform: str
    profile_status: Literal["pending", "running", "ready", "reused", "failed"]
    email_status: Literal["pending", "running", "available", "missing", "failed"]
    analysis_job_id: UUID | None
    profile_error_code: str | None
    email_error_code: str | None


class CreatorSearchPage(BaseModel):
    items: list[CreatorSearchView]
    total: int
    limit: int
    offset: int


class CreatorSearchUnitPage(BaseModel):
    items: list[CreatorSearchUnitView]
    total: int
    limit: int
    offset: int
